tests/Microsoft.Bot.Builder.Tests/RootFrameTests.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Microsoft.Bot.Builder.Tests
+{
+    [TestClass]
+    [TestCategory("State")]
+    public class RootFrameTests
+    {
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void Ctor_Should_Throw_On_NullStorage()
+        {
+            var x = new RootFrame(null, new FrameDefinition());
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void Ctor_Should_Throw_On_NullFrameDefinition()
+        {            
+            var x = new RootFrame(new Mock<IStorage>().Object, null);
```

Comment:
This pattern can be simplified. Instead of having to do `new Mock<Whatever>.Object` you can just write `Mock.Of<Whatever>()`.

NOTE: only putting this one comment on, but applies everywhere the pattern is being used.